gnd/src/main/java/com/google/android/gnd/ui/recorddetails/RecordDetailsViewModel.java

```diff
@@ -43,6 +47,21 @@ public void loadRecordDetails(String projectId, String featureId, String recordI
     disposeOnClear(
         dataRepository
             .getRecordDetails(projectId, featureId, recordId)
-            .subscribe(record::setValue));
+            .subscribe(this::onRecordUpdate));
```

Comment:
Rather than subscribing, while we're at it can we transform this into a LiveData<Integer> instead?

I think you just need to set lifecycle of the binding in the view with:
binding.setLifecycleOwner(this);

And in the constructor you would do something like:

```java
progressBarVisibility = 
LiveDataReactiveStreams.fromPublisher(
    dataRepository
            .getRecordDetails(projectId, featureId, recordId)
            .map(this::toProgressBarVisibility)
            .toFlowable(BackpressureStrategy.LATEST));
```

Then the view would fully manage the subscription based on its own lifecycle!

Reply:
Oh, and you'll also need to handle errors since LiveData doesn't:
```java
progressBarVisibility = 
LiveDataReactiveStreams.fromPublisher(
    dataRepository
            .getRecordDetails(projectId, featureId, recordId)
            .map(this::toProgressBarVisibility)
            .onErrorReturn(err -> Visibility.HIDDEN)
            .toFlowable(BackpressureStrategy.LATEST));
```
In the future we'll also want to report these errors, so perhaps a TODO would be in order.

Reply:
We'll see how this goes. Gave it a first pass: e678555f2 It also required me to use safeUnbox() in the layout to unbox the data--not sure whether or not that's something we should be doing. lmk your thoughts!